### PR TITLE
Ansible playbook added for knative & tekton

### PR DESCRIPTION
**Pull request by valen-mascarenhas14**
Added a separate Ansible playbook to install an NFS server, which is required for running tests for Knative and Tekton.

### CODE REVIEW

File: roles/install-nfs/files/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        - name: nfs-client-provisioner
+          image: quay.io/external_storage/nfs-client-provisioner:latest
+          volumeMounts:
+            - name: nfs-client-root
+              mountPath: /persistentvolumes
+          env:
+            - name: PROVISIONER_NAME
+              value: fuseim.pri/ifs
+            - name: NFS_SERVER
+              value: 10.10.10.60
```
Comment (mkumatag) [inline roles/install-nfs/files/deployment.yaml:32]: what is this ip address?

File: roles/install-nfs/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+- name: Start services
+  block:
+    - name: Start rpcbind
+      systemd:
+        name: rpcbind
+        enabled: yes
+        state: restarted
+
+    - name: Start nfs-server
```
Comment (mkumatag) [inline roles/install-nfs/tasks/main.yaml:28]: I was wondering if you want to server nfs-server also part of pod?!

Comment (mkumatag) [reply roles/install-nfs/tasks/main.yaml:28]: something similar to this - https://github.com/appscode/third-party-tools/blob/master/storage/nfs/README.md

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: roles/install-nfs/files/deployment.yaml (hunk 1 above)
Comment (valen-mascarenhas14) [reply roles/install-nfs/files/deployment.yaml:32]: @mkumatag 
The IP address `10.10.10.60` is a placeholder representing the private IP of the master node where the NFS server is set up. This value is a dummy and will be dynamically substituted with the actual IP.

You can find where it is referenced here: [deploy-nfs-client.sh.j2](https://github.com/valen-mascarenhas14/k8s-ansible/blob/0c4b1b126b0cdab648a7a311b452f02e312bb89c/roles/install-nfs/templates/deploy-nfs-client.sh.j2#L24):

File: roles/install-nfs/tasks/main.yaml (hunk 2 above)
Comment (valen-mascarenhas14) [reply roles/install-nfs/tasks/main.yaml:28]: The NFS server is set up on the master node, while the provisioner pod in Kubernetes acts as the client.

**Comment by valen-mascarenhas14** [comment]
@mkumatag  @Rajalakshmi-Girish  any issues with the PR ?

File: roles/install-nfs/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- name: Copy nfs client resource files to /tmp
+  copy:
+    src: "{{ item }}"
+    dest: /tmp/
+  with_fileglob:
+    - "*"
+
+- name: Template a nfs client script to /tmp/deploy-nfs-client.sh
+  template:
+    src: deploy-nfs-client.sh.j2
```
Comment (mkumatag) [inline roles/install-nfs/tasks/main.yaml:52]: I think we can move the content of this shell script to ansible playbook itself

File: roles/install-nfs/files/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            - name: NFS_PATH
+              value: /ifs/kubernetes
+      volumes:
+        - name: nfs-client-root
+          nfs:
+            server: 10.10.10.60
```
Comment (mkumatag) [inline roles/install-nfs/files/deployment.yaml:38]: use env to get the ip address where the pod is getting deployed, eg: https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables

File: roles/install-nfs/files/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+apiVersion: apps/v1
```
Comment (mkumatag) [inline roles/install-nfs/files/deployment.yaml:1]: convert this file to a template and replace the values like PATH, image etc.. via ansible variable.

File: roles/install-nfs/templates/deploy-nfs-client.sh.j2 [hunk 6]
```diff
@@ -0,0 +1,50 @@
+#! /bin/bash
```
Comment (mkumatag) [inline roles/install-nfs/templates/deploy-nfs-client.sh.j2:1]: let us avoid this script and convert to enable resources.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: roles/install-nfs/tasks/main.yaml (hunk 3 above)
Comment (valen-mascarenhas14) [reply roles/install-nfs/tasks/main.yaml]: Have mode it to ansible playbook

File: roles/install-nfs/files/deployment.yaml (hunk 5 above)
Comment (valen-mascarenhas14) [reply roles/install-nfs/files/deployment.yaml]: done

File: roles/install-nfs/templates/deploy-nfs-client.sh.j2 (hunk 6 above)
Comment (valen-mascarenhas14) [reply roles/install-nfs/templates/deploy-nfs-client.sh.j2]: done

File: roles/install-nfs/files/deployment.yaml (hunk 4 above)
Comment (valen-mascarenhas14) [reply roles/install-nfs/files/deployment.yaml]: done

**Comment by valen-mascarenhas14** [comment]
@mkumatag  Have made the changes you requested . Tried running it locally & it worked

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: group_vars/all [hunk 7]
```diff
@@ -79,3 +79,9 @@ etcd_version: v3.5.9
 cni_plugins_version: v1.3.0
 cni_plugins_url: https://github.com/containernetworking/plugins/releases/download
 cni_plugins_tarball: "cni-plugins-linux-{{ ansible_architecture }}-{{ cni_plugins_version }}.tgz"
+
+# NFS server details
+nfs_directory: "/var/nfsshare"
+
+# yq download url
+yq_download_url: "https://github.com/mikefarah/yq/releases/download/v4.40.5/yq_linux_ppc64le"
```
[the remark this answers is not in the document]
Comment (valen-mascarenhas14) [reply group_vars/all]: It's no longer used . Have removed it

File: roles/install-nfs/templates/nfs-client-provisioner.yaml.j2 [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      app: nfs-client-provisioner
+  template:
+    metadata:
+      labels:
+        app: nfs-client-provisioner
+    spec:
+      serviceAccountName: nfs-client-provisioner
+      containers:
+        - name: nfs-client-provisioner
+          image: {{ image | default('k8s.gcr.io/sig-storage/nfs-subdir-external-provisioner:v4.0.2') }}
```
[the remark this answers is not in the document]
Comment (valen-mascarenhas14) [reply roles/install-nfs/templates/nfs-client-provisioner.yaml.j2]: Modified it & removed the image variable

File: roles/install-nfs/templates/nfs-client-provisioner.yaml.j2 [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+              mountPath: /persistentvolumes
+          env:
+            - name: PROVISIONER_NAME
+              value: fuseim.pri/ifs
+            - name: NFS_SERVER
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.hostIP
+            - name: NFS_PATH
+              value: {{ nfs_directory | default('/var/nfsshare') }}
```
[the remark this answers is not in the document]
Comment (valen-mascarenhas14) [reply roles/install-nfs/templates/nfs-client-provisioner.yaml.j2]: Have removed it

File: roles/install-nfs/templates/nfs-client-provisioner.yaml.j2 [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+      volumes:
+        - name: nfs-client-root
+          nfs:
+            server: "{{ nfs_server }}"
+            path: {{ nfs_directory | default('/var/nfsshare') }}
```
[the remark this answers is not in the document]
Comment (valen-mascarenhas14) [reply roles/install-nfs/templates/nfs-client-provisioner.yaml.j2]: removed it

**Comment by valen-mascarenhas14** [comment]
@mkumatag any update on this ??

File: roles/install-nfs/tasks/main.yaml [hunk 11]
```diff
@@ -0,0 +1,88 @@
+---
+- name: Install prereq & nfs client packages
+  yum:
+    name: "{{ packages }}"
+    state: present
+  vars:
+    packages:
+      - wget
```
Comment (mkumatag) [inline roles/install-nfs/tasks/main.yaml:8]: where is this used `wget`?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: roles/install-nfs/tasks/main.yaml (hunk 11 above)
Comment (valen-mascarenhas14) [reply roles/install-nfs/tasks/main.yaml]: @mkumatag 
It was used to download yq, it's no longer needed since we updated the scipt. Have removed it.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**